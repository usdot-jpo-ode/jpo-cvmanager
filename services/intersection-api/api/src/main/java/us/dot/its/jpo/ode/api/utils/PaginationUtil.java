package us.dot.its.jpo.ode.api.utils;

import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

public class PaginationUtil {

    public static <T> ResponseEntity<List<T>> createResponseEntityWithPaginationHeaders(Page<T> page) {
        HttpHeaders headers = new HttpHeaders();
        headers.add("X-Total-Count", String.valueOf(page.getTotalElements()));
        headers.add("X-Total-Pages", String.valueOf(page.getTotalPages()));
        headers.add("X-Has-More", String.valueOf(page.hasNext()));
        return new ResponseEntity<List<T>>(page.getContent(), headers, HttpStatus.OK);
    }
}