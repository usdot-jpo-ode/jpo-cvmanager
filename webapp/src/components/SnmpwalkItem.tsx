import React from 'react'

import { Box, Button, Typography, useTheme } from '@mui/material'
import DeleteIcon from '@mui/icons-material/Delete'

export type SnmpwalkRowProps = {
  title: string
  value: string
}

export const SnmpWalkRow = (props: SnmpwalkRowProps) => {
  return (
    <div style={{ display: 'flex', flexDirection: 'row', alignItems: 'center', marginTop: -2 }}>
      <Typography sx={{ display: 'flex', alignItems: 'center', marginLeft: 1.5, lineHeight: 0.2 }}>
        <Typography component="span" color="textSecondary" sx={{ fontWeight: 'bold' }}>
          {props.title}
        </Typography>
        <Typography component="span" sx={{ paddingLeft: '8px' }}>
          {props.value}
        </Typography>
      </Typography>
    </div>
  )
}

export type SnmpwalkItemProps = {
  index: string
  content: any
  handleDelete: (countsMsgType: string, ip: string) => void
}

const SnmpwalkItem = (props: SnmpwalkItemProps) => {
  const theme = useTheme()

  const reformatDate = (ts: string) => {
    // ts like 2024-03-31T18:00:00-06:00
    const date = new Date(ts)
    return date.toISOString()
  }

  return (
    <Box
      id="snmpitemdiv"
      sx={{
        borderRadius: 2,
        padding: 2,
        backgroundColor: theme.palette.secondary.main,
        '&:hover': {
          backgroundColor: theme.palette.secondary.light,
        },
        height: '150px',
      }}
    >
      <h3 id="snmpitemheader">{props.index}</h3>
      <SnmpWalkRow title="Message Type:" value={props.content['Message Type']} />
      <SnmpWalkRow title="Destination IP:" value={props.content['IP']} />
      <SnmpWalkRow title="Port:" value={props.content['Port']} />
      <SnmpWalkRow title="Start:" value={reformatDate(props.content['Start DateTime'])} />
      <SnmpWalkRow title="End:" value={reformatDate(props.content['End DateTime'])} />
      <SnmpWalkRow title="Active:" value={props.content['Config Active']} />

      <Button
        sx={{ marginLeft: 20, marginTop: -4 }}
        onClick={() => props.handleDelete(props.content['Message Type'], props.content['IP'])}
        startIcon={<DeleteIcon />}
      >
        Delete
      </Button>
    </Box>
  )
}

export default SnmpwalkItem
