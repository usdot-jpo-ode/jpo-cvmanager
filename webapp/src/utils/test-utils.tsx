/**
 * This function searches for the every somewhat random/chaotic class name and property that cause snapshot tests to be inconsistent.
 * This current list includes MUI classes (css-*) and aria-invalid attributes.
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceChaoticIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container: HTMLElement) {
  const props = [
    {
      selector: 'class',
      updateFunc: (val: string) => val.replace(/css-[0-9a-z]*?-/g, 'css-mocked-'),
    },
  ]

  container.querySelectorAll('*').forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)))
      }
    })
  })
  container.querySelectorAll('input[aria-invalid]').forEach((item) => {
    item.removeAttribute('aria-invalid')
  })
  return container
}

import React from 'react'
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider'
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs'
import dayjs from 'dayjs'
import utc from 'dayjs/plugin/utc'
import timezone from 'dayjs/plugin/timezone'

// Extend dayjs with timezone and UTC plugins
dayjs.extend(utc)
dayjs.extend(timezone)

// Mock component to set timezone
const MockLocalizationProvider = ({ children }) => {
  dayjs.tz.setDefault('America/Denver')
  return <LocalizationProvider dateAdapter={AdapterDayjs}>{children}</LocalizationProvider>
}

export { replaceChaoticIds, MockLocalizationProvider }
