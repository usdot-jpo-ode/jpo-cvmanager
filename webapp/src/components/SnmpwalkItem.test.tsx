import React from 'react'
import { render } from '@testing-library/react'
import SnmpwalkItem from './SnmpwalkItem'
import { replaceChaoticIds } from '../utils/test-utils'

it('should take a snapshot', () => {
  const { container } = render(<SnmpwalkItem content={{}} index={''} handleDelete={() => ({})} />)

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
