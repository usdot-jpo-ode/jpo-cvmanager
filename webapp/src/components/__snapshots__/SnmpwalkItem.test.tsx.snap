// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-ppqe1x"
    id="snmpitemdiv"
  >
    <h3
      id="snmpitemheader"
    />
    <div
      style="display: flex; flex-direction: row; align-items: center; margin-top: -2px;"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        >
          Message Type:
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        />
      </p>
    </div>
    <div
      style="display: flex; flex-direction: row; align-items: center; margin-top: -2px;"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        >
          Destination IP:
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        />
      </p>
    </div>
    <div
      style="display: flex; flex-direction: row; align-items: center; margin-top: -2px;"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        >
          Port:
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        />
      </p>
    </div>
    <div
      style="display: flex; flex-direction: row; align-items: center; margin-top: -2px;"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        >
          Start:
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        />
      </p>
    </div>
    <div
      style="display: flex; flex-direction: row; align-items: center; margin-top: -2px;"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        >
          End:
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        />
      </p>
    </div>
    <div
      style="display: flex; flex-direction: row; align-items: center; margin-top: -2px;"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        >
          Active:
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        />
      </p>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-mocked-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-mocked-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mocked-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
          />
        </svg>
      </span>
      Delete
    </button>
  </div>
</div>
`;
